### ISSUE TITLE
fix(no-half-width-punctuation): --fix does not replace ASCII parentheses in mixed full-width/half-width case

### ISSUE THREAD

**Issue by luojiyin1987** [lint-md/lint-md#155]
## Summary
`@lint-md/core` reports `no-half-width-punctuation` correctly for ASCII parentheses in Chinese text, but `--fix` does not actually change the content in some mixed full-width/half-width parenthesis cases.

## Reproduction
Input line:

```md
在 DOM 检查器中按下 ctrl + \\（(Chrome/Window）) 可以随时暂停 JS 的执行。这样您就可以检查 DOM 的快照，而不必担心 JS 会改变 DOM 或事件（如鼠标悬停）会导致 DOM 从您脚下发生变化。
```

Run:

```bash
lint-md browser-debugging-tricks.md
lint-md --fix browser-debugging-tricks.md
lint-md browser-debugging-tricks.md
```

## Actual behavior
The rule reports two errors before fix:

- `(` at line 272, column 23
- `)` at line 272, column 38

But after `--fix`, the file content is unchanged and the same two errors are still reported.

## Expected behavior
`--fix` should rewrite the ASCII parentheses to full-width ones, so the line becomes:

```md
在 DOM 检查器中按下 ctrl + \\（（Chrome/Window）） 可以随时暂停 JS 的执行。这样您就可以检查 DOM 的快照，而不必担心 JS 会改变 DOM 或事件（如鼠标悬停）会导致 DOM 从您脚下发生变化。
```

## Notes
I checked the built rule implementation in `@lint-md/core@2.1.1`:

- rule file: `lib/rules/no-half-width-punctuation.js`
- fix is created with `fixer.replaceTextRange([pos.offset, pos.offset + 1], fullChar)`

The diagnostics point to the ASCII parentheses correctly, but the generated fix ranges appear to target the adjacent full-width parentheses instead.

For the reproduction line, the character positions on that line are:

- index 22: `（`
- index 23: `(`
- index 37: `）`
- index 38: `)`

But the generated fixes resolve to ranges that map to the already-full-width characters, so the applied replacement becomes a no-op.

## Environment
- `@lint-md/core`: `2.1.1`
- `@lint-md/cli`: `2.1.1`
